qa/integration-tests/src/test/java/io/camunda/it/backup/DataGenerator.java

```diff
@@ -13,118 +13,166 @@
                       .newProcessInstanceQuery()
                       .filter(
                           b ->
-                              b.processInstanceKey(p -> p.in(instancekeys.stream().toList()))
-                                  .state(ProcessInstanceState.COMPLETED))
-                      .page(b -> b.limit(instancekeys.size()).from(0))
+                              b.processInstanceKey(p -> p.in(instanceKeys.stream().toList()))
+                                  .state(state))
+                      .page(b -> b.limit(instanceKeys.size()).from(0))
                       .send();
               assertThat(response)
-                  .succeedsWithin(Duration.ofSeconds(timeoutSeconds))
+                  .succeedsWithin(timeout)
                   .extracting(SearchQueryResponse::items)
                   .asInstanceOf(InstanceOfAssertFactories.LIST)
-                  .hasSameSizeAs(instancekeys);
+                  .hasSameSizeAs(instanceKeys);
+
+              // remove all completed instances
+              response.get().items().stream()
+                  .filter(inst -> inst.getState().equals(ProcessInstanceState.COMPLETED))
+                  .map(ProcessInstance::getProcessInstanceKey)
+                  .forEach(instanceKeys::remove);
             });
   }
 
-  private void completeTask(final String jobType, final String varName, final int number) {
-    final var completedJobs = new AtomicInteger(0);
+  private void completeTask(final String jobType, final String varName) {
+    final var completedJobs = new ConcurrentHashMap<Long, Object>();
+    LOGGER.debug(
+        "Starting to {} complete tasks of type {} with processInstanceKeys={}",
+        instanceKeys.size(),
+        jobType,
+        instanceKeys);
     final var worker =
         camundaClient
             .newWorker()
             .jobType(jobType)
             .handler(
                 (jobClient, job) -> {
+                  LOGGER.debug(
+                      "Completing job {} of process {}", job.getKey(), job.getProcessInstanceKey());
                   jobClient
                       .newCompleteCommand(job.getKey())
                       .variable(varName, job.getKey())
                       .send()
-                      .thenRun(completedJobs::incrementAndGet);
+                      .join();
+                  completedJobs.put(job.getProcessInstanceKey(), job);
                 })
-            .timeout(5000)
-            .name("completeTask");
-    try (final var jobWorker = worker.open()) {
+            .timeout(timeout.toSeconds())
+            .name("completeTask")
+            .open();
+    try {
       Awaitility.await("until all jobs have been completed")
-          .atMost(Duration.ofSeconds(30))
-          .untilAtomic(completedJobs, Matchers.equalTo(number));
+          .atMost(timeout)
+          .untilAsserted(() -> assertThat(completedJobs).hasSize(instanceKeys.size()));
+    } finally {
+      worker.close();
```

Comment:
:wrench: Maybe it makes sense to simply send an activate commamd to query for the jobs and complete them directly without using a worker because it seems to be a bit unreliable

Reply:
Thanks for the input, it's much simpler :+1: